Workspaces/Base/WorkspaceUI.cs

```diff
@@ -444,6 +451,185 @@ IEnumerator Start()
 			yield return null;
 		}
 
+		void AdjustHandlesAndIcons()
```

Comment:
This method looks like a lot of repeated/similar code. Any chance you can simply set up an array of constants/offsets and loop over that to handle this update?